app/models/concerns/course/levels_concern.rb

```diff
@@ -41,4 +41,35 @@ def compute_level_number(experience_points)
     level = compute_level(experience_points)
     level ? level.level_number : 0
   end
+
+  # Returns the current level's experience point (exp) threshold for
+  # for a given number of experience_points.
```

Comment:
That makes sense, to do that requires every course to have a default level where the `points_threshold` is 0, so that `current_course_user.level` will always return a level object. I'd think this would be the most intuitive way to do it (except that we have to initialise a default levels object upon course creation).

I've tried to contain all the messiness of the level's class inside `levels_concern` and arrived at this >_<